src/goto-programs/interpreter_evaluate.cpp

```diff
@@ -339,11 +339,19 @@ void interpretert::evaluate(
         evaluate(expr.op0(), dest);
         return;
       }
-      if(expr.has_operands() && !to_integer(expr.op0(), i))
+      else if(
+        expr.has_operands() && !to_integer(to_constant_expr(expr.op0()), i))
```

Comment:
What does a pointer-typed constant (mind the test in line 300) with operands even look like?

Reply:
Constant memory offset? `void *x = 0x100`? Don't know if that is _ever_ what you want

Reply:
@tautschnig this for example ```java::@inflight_exception
constant
  * type: pointer
      * width: 64
      * #no_nondet_initialization: 1
      0: empty
  * value: 0000000000000000000000000000000000000000000000000000000000000000
  0: constant
      * type: pointer
          * width: 64
          * #no_nondet_initialization: 1
          0: empty
      * value: NULL```
effectively this here was the culprit for the `failed to evaluate expression: null`:
cprover_string_content::cprover_string_content
constant
  * type: pointer
      * width: 64
      0: unsignedbv
          * width: 16
  * value: NULL
  * #source_location: 
    * file: org/apache/tika/parser/prt/PRTParser.java
    * line: 222
    * function: java::org.apache.tika.parser.prt.PRTParser.extractText:([BZ)Ljava/lang/String;
    * java_bytecode_index: 22
!! failed to evaluate expression: null
```
(both first name then `.pretty` output)

Reply:
That is definitely wrong (!) Could you raise a ticket either on issues here or on Jira to track this down. @LAJW I bet this is why you had some weird edge case handling code for null pointers!